src/server/scripts/Northrend/VioletHold/boss_moragg.cpp

```diff
@@ -90,25 +96,28 @@ class boss_moragg : public CreatureScript
             }
         }
 
-        void MoveInLineOfSight(Unit* /*who*/) override { }
-
         void UpdateAI(uint32 diff) override
         {
             if (!UpdateVictim())
                 return;
 
-            if (uiOpticLinkTimer <= diff)
-            {
-                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 100, true))
-                    DoCast(target, SPELL_OPTIC_LINK);
-                uiOpticLinkTimer = 15000;
-            } else uiOpticLinkTimer -= diff;
+            events.Update(diff);
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
 
-            if (uiCorrosiveSalivaTimer <= diff)
+            switch (uint32 eventId = events.ExecuteEvent())
```

Comment:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_moragg.cpp: In member function ‘virtual void boss_moragg::boss_moraggAI::UpdateAI(uint32)’:
/home/jackpoz/trinity/sources/src/server/scripts/Northrend/VioletHold/boss_moragg.cpp:111:28: warning: unused variable ‘eventId’ [-Wunused-variable]
             switch (uint32 eventId = events.ExecuteEvent())
                            ^